sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -540,32 +540,45 @@ private static boolean isPortResource(Protos.Resource resource) {
         return getNewExecutorInfo(podInstance.getPod(), serviceName, targetConfigurationId);
     }
 
-    private static Protos.ContainerInfo getContainerInfo(ContainerSpec containerSpec) {
+    private static Protos.ContainerInfo getContainerInfo(ContainerSpec containerSpec,
+                                                         Collection<NetworkSpec> networks) {
         Protos.ContainerInfo.Builder containerInfo = Protos.ContainerInfo.newBuilder()
                 .setType(Protos.ContainerInfo.Type.MESOS);
 
-        if (containerSpec.getImageName().isPresent()) {
+        if (containerSpec != null && containerSpec.getImageName().isPresent()) {
```

Comment:
Since optionality in the `containerSpec` parameter is already implicitly a part of this method's contract, why not just have it be of type `Optional<ContainerSpec>`? Saves an unwrapping in the caller that would need to happen here anyway in a different form via the `null` check.